sqlmesh/core/snapshot/definition.py

```diff
@@ -1040,12 +1018,179 @@ class SnapshotTableCleanupTask(PydanticModel):
 SnapshotNameVersionLike = t.Union[SnapshotNameVersion, SnapshotTableInfo, Snapshot]
 
 
-def table_name(physical_schema: str, name: str, version: str, is_temp: bool = False) -> str:
+class DeployabilityIndex(PydanticModel, frozen=True):
+    """Contains information about deployability of every snapshot.
+
+    Deployability is defined as whether or not the output that a snapshot produces during the
+    current evaluation can be reused in (deployed to) the production environment.
+    """
+
+    deployable_ids: t.Optional[t.FrozenSet[str]] = None
+    non_deployable_ids: t.Optional[t.FrozenSet[str]] = None
```

Comment:
Could this be simplified to be a single non-optional `FrozenSet[str]` and then a flag that says if it is deployable or non-deployable?